pkg/instancetype/instancetype.go

```diff
@@ -522,17 +523,50 @@ func (m *methods) findClusterInstancetype(vm *virtv1.VirtualMachine) (*instancet
 	return instancetype, nil
 }
 
+func resolveConflicts(field *k8sfield.Path, requestType k8sv1.ResourceName, vmiSpec *virtv1.VirtualMachineInstanceSpec, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) (Conflicts, bool) {
+	hasGuestField := false
+
+	switch requestType {
+	case k8sv1.ResourceCPU:
+		if condition := instancetypeSpec.CPU.Guest; condition != 0 {
```

Comment:
You need to check that `instancetypeSpec.CPU != nil` if it's going to be optional before checking `Guest`.

Also why are you pulling `condition` out here?